File: livingrimoire start here/LivinGrimoire java/src/Auxiliary_Modules/UniqueItemSizeLimitedPriorityQueue.java
package Auxiliary_Modules;

import java.util.ArrayList;

public class UniqueItemSizeLimitedPriorityQueue extends UniqueItemsPriorityQue{
    // items in the queue are unique and do not repeat
    // the size of the queue is limited
    private int limit = 5;

    public int getLimit() {
        return limit;
    }

    public void setLimit(int limit) {
        this.limit = limit;
    }

    @Override
    public void add(String item) {
        if(super.size() == limit){
            super.poll();}
        super.add(item);
    }

    @Override
    public String poll() {
        String temp = super.poll();
        if (temp == null){
            return "";
        }
        return temp;
    }

    @Override
    public String getRNDElement() {
        String temp = super.getRNDElement();
        if (temp == null){
            return "";
        }
        return temp;
    }

    public ArrayList<String> getAsList(){
        return getElements();
    }
}

File: livingrimoire start here/LivinGrimoire Kotlin/kotlin/auxiliary_modules/UniqueItemSizeLimitedPriorityQueue.kt
package auxiliary_modules

open class UniqueItemSizeLimitedPriorityQueue : UniqueItemsPriorityQue() {
    // items in the queue are unique and do not repeat
    // the size of the queue is limited
    var limit = 5

    override fun add(item: String) {
        if (super.size() == limit) {
            super.poll()
        }
        super.add(item)
    }

    override fun poll(): String {
        return super.poll() ?: return ""
    }

    override val rNDElement: String
        get() = super.rNDElement ?: ""
    val asList: ArrayList<String>
        get() = elements
}